eng/common/scripts/job-matrix/samples/matrix-test.yml

```diff
@@ -18,7 +18,7 @@ jobs:
         # Exclusion example
         - OSVmImage=^(?!macOS).*
       MatrixReplace:
-        - OsVmImage=.*ubuntu.*/ubuntu-20.04
+        - OsVmImage=(ubuntu).*/$1-20.04
```

Comment:
Might be worth adding a test case for the pool case.

Reply:
There are tests for regex grouping replacements: https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/job-matrix/tests/job-matrix-functions.modification.tests.ps1#L513